### ISSUE TITLE
Integrate with Viewer

### ISSUE THREAD

**Issue by ryanccarelli** [Dana-Farber-AIOS/pathml#174]
The multiplexed MIF workflow requires actively moving back and forth between raw image and phenotype plots to choose optimal phenotype thresholds and check that phenotypes match the pathologist's expectations. At the moment users will have images open with Fiji or Halo and export the cell centroids (colored by phenotype) into this software. Ideally this workflow would be fully self-contained in pathML. To accomplish this we need to support a viewer. 

One promising open source option is Napari https://napari.org/index.html

I wanted to open a discussion here since there has been significant interest in this feature

**Comment by dmbrundage** [comment]
Is the idea to integrate Napari for both MIF workflows and the H&E workflow as well? I see that there has been attempts to integrate Napari with OpenSlide.

**Comment by ryanccarelli** [comment]
Yes I think that we should be able to open the image and masks of any SlideData object in a viewer.

Thanks pointing me to the OpenSlide efforts, I found https://github.com/manzt/napari-lazy-openslide did you have any other projects in mind?

**Comment by dmbrundage** [comment]
No, that is the project I was referencing but haven't really dug to deep into anything. OpenSeaDragon is another popular view that I see referenced (https://openseadragon.github.io/). 

The Cancer Digital Slide Archive is built on top of OpenSeaDragon. I have also built a small react app that uses OpenSeaDragon as a viewer as well.

https://cancer.digitalslidearchive.org/#!/CDSA/acc/TCGA-OR-A5J1/TCGA-OR-A5J1-01A-01-TS1

**Comment by dmbrundage** [comment]
Here are a couple more projects I have found:

https://kaibu.org/#/app
http://vitessce.io/?dataset=linnarsson-2018&theme=dark (https://github.com/vitessce/vitessce)
http://avivator.gehlenborglab.org/?image_url=https://viv-demo.storage.googleapis.com/brain.pyramid.ome.tif

Looking at Napari I think it probably does make a lot of sense for integration into PathML since it is running PyQT locally.

I have only been able to get OpenSeaDragon functioning with DeepZoom format. I believe OpenSlide has a DeepZoom generator and conversion script but this is only ideal for web based browsing which is probably not necessary if the viewer will only be used locally as part of the workflow.

**Comment by jacobrosenthaldfci** [comment]
Another viewer is [Minerva](https://www.cycif.org/software/minerva) which looks like it's also built on Openseadragon
